src/snowflake/snowpark/dataframe.py

```diff
@@ -508,8 +509,10 @@ def __init__(
         session: Optional["snowflake.snowpark.Session"] = None,
         plan: Optional[LogicalPlan] = None,
         is_cached: bool = False,
```

Comment:
I'd like to understand a bit better (for me) what ast_smt is supposed to be. Can we either add a comment, or maybe a simple test - just to see how it's supposed to be used?